x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/tree_table.tsx

```diff
@@ -24,67 +49,217 @@ export function StreamsTreeTable({
+              }
+            );
+            counts[name] = details?.count ?? 0;
+          } catch (e) {
+            if (!controller.signal.aborted) {
+              counts[name] = 0;
+            }
+          }
+        })
+      );
+      if (!controller.signal.aborted) {
+        setDocCounts(counts);
+      }
+    };
+    fetchCounts();
+    return () => {
+      controller.abort();
+    };
+  }, [streamNames, timeState, streamsRepositoryClient]);
+  type SortableField = 'nameSortKey' | 'documentsCount' | 'retentionMs';
+  const [sortField, setSortField] = useState<SortableField>('nameSortKey');
+  const [sortDirection, setSortDirection] = useState<Direction>('asc');
+
+  const enrichedTree = React.useMemo(() => {
+    const enrich = (node: StreamTree): EnrichedStreamTree => {
+      const documentsCount = docCounts[node.name] ?? 0;
+      let retentionMs = 0;
+      const lc = node.effective_lifecycle;
+      if (isDslLifecycle(lc)) {
+        retentionMs = parseDurationInSeconds(lc.dsl.data_retention ?? '') * 1000;
+      } else if (isIlmLifecycle(lc)) {
+        retentionMs = Number.POSITIVE_INFINITY;
+      }
+      const nameSortKey = `${getSegments(node.name).length}_${node.name.toLowerCase()}`;
+      return {
+        ...node,
+        nameSortKey,
+        documentsCount,
+        retentionMs,
+        children: node.children.map(enrich),
+      } as EnrichedStreamTree;
+    };
+    return asTrees(streams ?? []).map(enrich);
+  }, [streams, docCounts]);
+
+  const flattenTree = useCallback((roots: EnrichedStreamTree[]): TableRow[] => {
+    const result: TableRow[] = [];
+    const walk = (node: EnrichedStreamTree, level: number, rootRow: TableRow) => {
+      node.children.forEach((child) => {
+        const childRow: TableRow = {
+          ...child,
+          level,
+          rootNameSortKey: rootRow.rootNameSortKey,
+          rootDocumentsCount: rootRow.rootDocumentsCount,
+          rootRetentionMs: rootRow.rootRetentionMs,
+        };
+        result.push(childRow);
+        walk(child, level + 1, rootRow);
+      });
+    };
+    roots.forEach((root) => {
+      const rootRow: TableRow = {
+        ...root,
+        level: 0,
+        rootNameSortKey: root.nameSortKey,
+        rootDocumentsCount: root.documentsCount,
+        rootRetentionMs: root.retentionMs,
+      };
+      result.push(rootRow);
+      walk(root, 1, rootRow);
+    });
+    return result;
+  }, []);
+
+  const sortedRoots = React.useMemo(() => {
+    return [...enrichedTree].sort((a, b) => {
+      const av = a[sortField];
+      const bv = b[sortField];
+      if (typeof av === 'string' && typeof bv === 'string') {
+        return sortDirection === 'asc' ? av.localeCompare(bv) : bv.localeCompare(av);
+      }
+      if (typeof av === 'number' && typeof bv === 'number') {
+        return sortDirection === 'asc' ? av - bv : bv - av;
+      }
+      return 0;
+    });
+  }, [enrichedTree, sortField, sortDirection]);
```

Comment:
The sorting logic does not work correctly. I think in this case "logs.linux" should come before "logs.apache":

![Screenshot 2025-06-17 at 14 30 51](https://github.com/user-attachments/assets/06848ebc-6874-47d6-a1b7-9f061a4ac1a5)

Reply:
Right now it should work

Reply:
Yeh but I think the sort order is inverted. Ascending means 0-9/A-Z but here it seems to be the other way around:

<img width="938" alt="Screenshot 2025-06-19 at 09 00 11" src="https://github.com/user-attachments/assets/4de3daad-a1a2-4eb6-8038-3d392243fae0" />

Can you also add unit tests for the sorting logic to ensure it works correctly?